pkg/apis/intevents/v1alpha1/brokercell_types.go

```diff
@@ -71,6 +71,24 @@ var (
 // ComponentParameters specifies scaling and resource parameters to be used
 // by a single component of a BrokerCell.
 type ComponentParameters struct {
+	// AvgCPUUtilization specifies the average CPU consumption targeted by the component's Horizontal Pod Autoscaler
+	AvgCPUUtilization *int32 `json:"avgCPUUtilization,omitempty"`
+
+	// AvgMemoryUsage specifies the average memory consumption targeted by the component's Horizontal Pod Autoscaler
+	AvgMemoryUsage *string `json:"avgMemoryUsage,omitempty"`
+
+	// CPURequest specifies the minimal amount of the CPU for the deployment to be schedulable
+	CPURequest *string `json:"cpuRequest,omitempty"`
+
+	// CPULimit specifies the maximal amount of the CPU to be consumable by the deployment
+	CPULimit *string `json:"cpuLimit,omitempty"`
+
+	// MemoryRequest specifies the minimal amount of memory for the deployment to be schedulable
+	MemoryRequest *string `json:"memoryRequest,omitempty"`
+
+	// MemoryLimit specifies the maximal amount of memory to be consumable by the deployment
+	MemoryLimit *string `json:"memoryLimit,omitempty"`
+
```

Comment:
Does it make sense to directly use `corev1.ResourceRequirements` here?

Reply:
It should be doable and I see value in doing that. It also seems natural to separate input params from the requirements structure, especially that we've got a mix of autoscaling and resource params here. I guess the main consideration here is how the product and that internal structure may evolve over time. A typical example would be: if there will be an update in the format of the internal structure, during upgrade, this may possibly require an update of existing deployment yamls (i.e. customer side), whereas by having those separated, we may limit the change to the product update (i.e. our side).